pkg/reconciler/revision/resources/deploy.go

```diff
@@ -257,11 +263,18 @@ func MakeDeployment(rev *v1.Revision,
 		return nil, fmt.Errorf("failed to create PodSpec: %w", err)
 	}
 
+	labels := makeLabels(rev)
+
+	if istioConfig != nil {
```

Comment:
This check should be against the configured ingress class. This can be configured by either annotation or by configmap) 

* Configmap: https://github.com/knative/serving/blob/master/config/core/configmaps/network.yaml#L86
* Annotation Key: https://github.com/knative/serving/blob/a926087ab524b37e42b15222332af8c812adc614/pkg/apis/networking/register.go#L36

You already have networkConfig passed in; however, checking the value of the annotation does not make sense from this location. For this reason this logic will need to be moved out of MakeDeployment to somewhere else. 

Two options I can think of here:
(1) Up into the Service reconciler logic since it contains both the Route (and thus ingress) as well as the Configuration (and thus Deployment) resources.
(2) Injected into the Deployment post-creation based upon how the Route and Configuration and coupled.